### ISSUE TITLE
Websocket callbacks are not propely cleaned up

### ISSUE THREAD

**Issue by voyagerok** [strophe/strophejs#318]
Is there a reason why Strophe.Websocket._closeSocket() cleans up only error callback and others stay untouched?
```
_closeSocket: function () {
        if (this.socket) {
            try {
                this.socket.onerror = null;
                this.socket.close();
            } catch (e) {
                Strophe.debug(e.message);
            }
        }
        this.socket = null;
},
```
This can lead to nasty situations, when we receive events from old sockets, but handle them using newly created one.

**Comment by voyagerok** [comment]
I came across a problem with code, which have two subsequent calls to Strophe.Connection.connect() with a very small period of time (not intentionally though). Second call did not clean up old socket properly and in some situations this zombie socket generated open event, and callback tried to send data on new socket, which was not properly initialized by that time. I believe there are other scenarios in which this problem can fire.

**Comment by sualko** [comment]
I'm not familiar with the websocket code, but it sounds like a bug. It would be awesome if you could provide a pull request to fix this.

**Comment by voyagerok** [comment]
Closed with 31f31b5

**Comment by jcbrand** [comment]
@voyagerok Thanks for coming back to close this ticket.
